### ISSUE TITLE
Background for import-maps integrity

### ISSUE THREAD

**Issue by tomrittervg** [guybedford/import-maps-extensions#23]
I'm getting up to speed on several standards-related things, and am reading about import-maps integrity.  It makes sense to me (I think) but I had a couple of questions.  It says

> Without a mechanism to specify integrity, it is not currently possible to use module dependencies with require-sri-for Content Security Policy where those module dependencies are loaded lazily so that the integrity cannot be set via the module script tag or link preload tag directly.

I couldn't find `require-sri-for` in CSP, and it seems like this proposal got backed out/unimplemented, is that correct?  Is it being actively revived in this form or an analogous one?

**Comment by hybrist** [comment]
I think that sentence applies to a "strict, hash-based CSP" ([MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP#strict_csp)). At least my understanding is that it would require the use of `strict-dynamic` to implicitly trust all modules loaded from the initial module(s).

**Comment by tomrittervg** [comment]
Okay, thanks, I'm still getting back up to speed on all the intricacies of things, that technique seems reasonable but it feels like there's still a hole; no?

If you specify a strict, hash-based CSP, you need to either specify all the hashes of the modules you import using import-maps _or_ use strict-dynamic to allow those modules through.  The first is a strict allow-list so that's safe.  The second will rely on the import map's integrity definitions to validate them.  _But_ it doesn't seem there's a mechanism to require all imports in an import-map to have an integrity attribute, and if they don't have one, abort the load - is that right?

**Comment by sambacha** [comment]
No, all imports in an import-map do not have to have an integrity attribute.

The integrity attribute in import maps is completely optional and can be applied selectively. This flexibility is intentional in the design of the import map specification to support different use cases and security requirements.

Edit: forgot to include that chrome does have a feature flag to enforce this behaviour for requiring sri digest:

https://chromestatus.com/feature/5090023365672960
